libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -212,6 +218,100 @@ class MapboxNavigationTest {
         verify(exactly = 0) { fasterRouteObserver.onFasterRouteAvailable(routes[0]) }
     }
 
+    @Test
+    fun reRoute_called() {
+        val offRouteObserverSlot = slot<OffRouteObserver>()
+        verify { tripSession.registerOffRouteObserver(capture(offRouteObserverSlot)) }
+
+        offRouteObserverSlot.captured.onOffRouteStateChanged(true)
+
+        val optionsSlot = slot<RouteOptions>()
+        verify(exactly = 1) { directionsSession.requestRoutes(capture(optionsSlot), any()) }
+    }
+
+    @Test
+    fun reRoute_called_with_null_bearings() {
+        val routeOptions = provideDefaultRouteOptionsBuilder()
+                .coordinates(listOf(
+                        Point.fromLngLat(1.0, 1.0),
+                        Point.fromLngLat(1.0, 1.0),
+                        Point.fromLngLat(1.0, 1.0),
+                        Point.fromLngLat(1.0, 1.0)))
+                .build()
+
+        every { directionsSession.getRouteOptions() } returns routeOptions
+
+        val offRouteObserverSlot = slot<OffRouteObserver>()
+        verify { tripSession.registerOffRouteObserver(capture(offRouteObserverSlot)) }
+
+        offRouteObserverSlot.captured.onOffRouteStateChanged(true)
+
+        val optionsSlot = slot<RouteOptions>()
+        verify(exactly = 1) { directionsSession.requestRoutes(capture(optionsSlot), any()) }
+
+        val options = optionsSlot.captured
+        val bearings = options.bearingsList()
+        val coordinates = options.coordinates()
+
+        assertEquals(coordinates.size, bearings!!.size)
```

Comment:
Open for discussion.

Generally is preferable having only 1 assertion per test 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=57

The problem of having multiple assertions are:
- It's annoying to have to look at the stacktrace to know what's failing (which assertion failed?)
- You don't have information about the remaining assertions (because when an assertion fails it breaks the execution of the test)

That's why is preferable to have only 1 assertion per test and one test per scenario. Basically, if one of them fails you'll know at a glance what in the code might have caused that failure and in which part (class, method, etc.).

I rarely write tests with multiple assertions (at least I try 😅), only if they're tearing apart the result object to verify everything e.g. if there's no `equals` (which is another problem although could happen if it's coming from a 3rd party library object) or if the assertions are atomic to the object (what we're asserting needs to be verified as whole). Is this one of these cases?

The same thing applies to the rest of the tests 👇

Reply:
fixed, compare expected and actual lists